packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -894,14 +906,29 @@ export function makeVirtualObjectManager(
   function defineDurableKindMulti(kindHandle, init, behavior, options) {
     assert(kindHandleToID.has(kindHandle), `unknown handle ${kindHandle}`);
     const kindID = kindHandleToID.get(kindHandle);
-    const durableKindDescriptor = kindIDToDescriptor.get(kindID);
-    assert(durableKindDescriptor);
-    const { tag, nextInstanceID } = durableKindDescriptor;
+    const descriptor = kindIDToDescriptor.get(kindID);
+    assert(descriptor);
+    const { tag, nextInstanceID, unfaceted, facets } = descriptor;
     assert(
       !definedDurableKinds.has(kindID),
       `redefinition of durable kind "${tag}"`,
     );
+    assert(
+      !unfaceted,
+      `durable kind "${tag}" originally defined as single-faceted`,
+    );
     nextInstanceIDs.set(kindID, nextInstanceID);
+    const definedFacets = Object.getOwnPropertyNames(behavior).sort();
```

Comment:
I'm a tiny bit nervous about having two copies of this facet-extraction/ordering line (and what happens if this one diverges from the other copy in `defineKindInternal()`).

What do you think about passing the descriptor into `defineKindInternal`, and letting `defineKindInternal` do both the comparison/assert and the subsequent `saveDurableKindDescriptor` (if `durable` and it was updated with facet info)?

In that approach, the descriptor argument is basically describing the facet names that have been committed to, so a future version of `defineKindInternal` has the option to append new facet names to that list, and it's in complete control of the ordering it picks.

Reply:
Yeah, I had the same misgiving, but I couldn't see a graceful way to deal with it that wasn't worse.  However, I've now stared at the code for a few minutes after not having looked at it for a day and I think I see a way to handle this more cleanly.